packages/alea-frontend/components/CoverageForm.tsx

```diff
@@ -122,8 +125,26 @@ export function CoverageForm({
     setFormData({ ...formData, isQuizScheduled: e.target.checked });
   };
 
+  const handleCheckboxSection = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFormData({ ...formData, sectionCompleted: e.target.checked });
+  };
+
   const handleSlideUriChange = (uri: string | undefined, slideNumber: number | undefined) => {
-    setFormData({ ...formData, slideUri: uri, slideNumber });
+    if (!formData.sectionUri || slideNumber === undefined || !uri) return;
+
+    const section = secInfo[formData.sectionUri];
+    if (!section) return;
+
+    getSlides(courseId, section.id).then(({ slides }) => {
```

Comment:
this is not a cached call. are you fetching slides of a section on every slide change? This component already fetches slides on section change. Perhaps you should store it in some state.